File: Shoppy/src/app/features/manager/model/Feedback.ts
import { Order } from './order.model';
import { Product } from './Product.model';
import { User } from './user';

export interface Feedback {
  id: number;
  productId: number;
  userId: string;
  orderId?: number;
  rating: number;
  comment: string;
  imageUrl?: string;
  createdDate: Date;

  product?: Product;
  user?: User;
  order?: Order;
}

File: ProjectSaleStore/ProjectSaleStore/Models/Feedback.cs
namespace SaleStoreDecor.Models
{
    public class Feedback
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public string UserId { get; set; }
        public int? OrderId { get; set; } 
        public int Rating { get; set; }
        public string Comment { get; set; }
        public string? ImageUrl { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;

        public Product Product { get; set; }
        public User User { get; set; }
        public Order Order { get; set; }
    }
}
